Marek Mahut <mmahut@fedoraproject.org>: Fix problems with compilation under gcc 4.3

diff -Naur celestia-1.5.0-upstream/src/buildstardb.cpp celestia-1.5.0/src/buildstardb.cpp
--- celestia-1.5.0-upstream/src/buildstardb.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/buildstardb.cpp	2008-03-01 08:52:23.000000000 +0100
@@ -17,6 +17,7 @@
 #include <assert.h>
 #include <unistd.h>
 #include "celengine/stardb.h"
+#include <libintl.h>                                                                                                         
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/asterism.cpp celestia-1.5.0/src/celengine/asterism.cpp
--- celestia-1.5.0-upstream/src/celengine/asterism.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/asterism.cpp	2008-03-01 09:23:23.000000000 +0100
@@ -8,6 +8,7 @@
 // of the License, or (at your option) any later version.
 
 #include <algorithm>
+#include <libintl.h>
 
 #ifndef _WIN32
 #ifndef TARGET_OS_MAC
diff -Naur celestia-1.5.0-upstream/src/celengine/astro.cpp celestia-1.5.0/src/celengine/astro.cpp
--- celestia-1.5.0-upstream/src/celengine/astro.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/astro.cpp	2008-03-01 09:14:33.000000000 +0100
@@ -10,12 +10,14 @@
 #include <cmath>
 #include <iomanip>
 #include <cstdio>
+#include <cstring>
 #include <time.h>
 #include <celutil/basictypes.h>
 #include <celmath/mathlib.h>
 #include "celestia.h"
 #include "astro.h"
 #include <celutil/util.h>
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/body.cpp celestia-1.5.0/src/celengine/body.cpp
--- celestia-1.5.0-upstream/src/celengine/body.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/body.cpp	2008-03-01 09:34:10.000000000 +0100
@@ -14,6 +14,7 @@
 #include <celmath/mathlib.h>
 #include <celutil/util.h>
 #include <celutil/utf8.h>
+#include <libintl.h>
 #include "mesh.h"
 #include "meshmanager.h"
 #include "body.h"
diff -Naur celestia-1.5.0-upstream/src/celengine/cmdparser.cpp celestia-1.5.0/src/celengine/cmdparser.cpp
--- celestia-1.5.0-upstream/src/celengine/cmdparser.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/cmdparser.cpp	2008-03-01 08:52:32.000000000 +0100
@@ -30,6 +30,7 @@
 #include "astro.h"
 #include "cmdparser.h"
 #include "glcontext.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/console.cpp celestia-1.5.0/src/celengine/console.cpp
--- celestia-1.5.0-upstream/src/celengine/console.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/console.cpp	2008-03-01 08:50:23.000000000 +0100
@@ -10,6 +10,7 @@
 #include <cstdarg>
 #include <cstdio>
 #include <cassert>
+#include <cstring>
 #include "celutil/utf8.h"
 #include "gl.h"
 #include "vecgl.h"
diff -Naur celestia-1.5.0-upstream/src/celengine/dds.cpp celestia-1.5.0/src/celengine/dds.cpp
--- celestia-1.5.0-upstream/src/celengine/dds.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/dds.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -15,6 +15,7 @@
 #include <celengine/gl.h>
 #include <celengine/glext.h>
 #include <celengine/image.h>
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/dsodb.cpp celestia-1.5.0/src/celengine/dsodb.cpp
--- celestia-1.5.0-upstream/src/celengine/dsodb.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/dsodb.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -32,6 +32,7 @@
 #include <celengine/galaxy.h>
 #include <celengine/opencluster.h>
 #include <celengine/nebula.h>
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/fragmentprog.cpp celestia-1.5.0/src/celengine/fragmentprog.cpp
--- celestia-1.5.0-upstream/src/celengine/fragmentprog.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/fragmentprog.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -13,6 +13,7 @@
 #include "gl.h"
 #include "glext.h"
 #include "fragmentprog.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/galaxy.cpp celestia-1.5.0/src/celengine/galaxy.cpp
--- celestia-1.5.0-upstream/src/celengine/galaxy.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/galaxy.cpp	2008-03-01 11:52:10.000000000 +0100
@@ -23,6 +23,8 @@
 #include "vecgl.h"
 #include "render.h"
 #include "texture.h"
+#include <libintl.h>
+#include <string.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/glcontext.cpp celestia-1.5.0/src/celengine/glcontext.cpp
--- celestia-1.5.0-upstream/src/celengine/glcontext.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/glcontext.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -12,6 +12,7 @@
 #include "gl.h"
 #include "glext.h"
 #include "glcontext.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/glext.cpp celestia-1.5.0/src/celengine/glext.cpp
--- celestia-1.5.0-upstream/src/celengine/glext.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/glext.cpp	2008-03-01 08:52:56.000000000 +0100
@@ -8,6 +8,7 @@
 // of the License, or (at your option) any later version.
 
 #include <string.h>
+#include <cstring>
 #include "gl.h"
 
 #ifndef _WIN32
@@ -19,6 +20,7 @@
 #endif /* ! _WIN32 */
 
 #include "glext.h"
+#include <libintl.h>
 
 // ARB_texture_compression
 glx::PFNGLCOMPRESSEDTEXIMAGE3DARBPROC glx::glCompressedTexImage3DARB;
diff -Naur celestia-1.5.0-upstream/src/celengine/image.cpp celestia-1.5.0/src/celengine/image.cpp
--- celestia-1.5.0-upstream/src/celengine/image.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/image.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -66,6 +66,7 @@
 #include <iostream>
 #include <algorithm>
 #include <cmath>
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/location.cpp celestia-1.5.0/src/celengine/location.cpp
--- celestia-1.5.0-upstream/src/celengine/location.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/location.cpp	2008-03-01 09:34:18.000000000 +0100
@@ -11,6 +11,7 @@
 #include <celengine/location.h>
 #include <celengine/body.h>
 #include <celutil/util.h>
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/lodspheremesh.cpp celestia-1.5.0/src/celengine/lodspheremesh.cpp
--- celestia-1.5.0-upstream/src/celengine/lodspheremesh.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/lodspheremesh.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -17,6 +17,7 @@
 #include "glext.h"
 #include "vecgl.h"
 #include "lodspheremesh.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/marker.cpp celestia-1.5.0/src/celengine/marker.cpp
--- celestia-1.5.0-upstream/src/celengine/marker.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/marker.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -9,6 +9,7 @@
 
 #include "marker.h"
 #include "gl.h"
+#include <libintl.h>
 
 
 using namespace std;
diff -Naur celestia-1.5.0-upstream/src/celengine/mesh.cpp celestia-1.5.0/src/celengine/mesh.cpp
--- celestia-1.5.0-upstream/src/celengine/mesh.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/mesh.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -14,6 +14,7 @@
 #include <cassert>
 #include <iostream>
 #include <algorithm>
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/meshmanager.cpp celestia-1.5.0/src/celengine/meshmanager.cpp
--- celestia-1.5.0-upstream/src/celengine/meshmanager.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/meshmanager.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -26,6 +26,7 @@
 #include "spheremesh.h"
 #include "texmanager.h"
 #include "meshmanager.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/modelfile.cpp celestia-1.5.0/src/celengine/modelfile.cpp
--- celestia-1.5.0-upstream/src/celengine/modelfile.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/modelfile.cpp	2008-03-01 11:52:21.000000000 +0100
@@ -15,6 +15,8 @@
 #include <cassert>
 #include <cmath>
 #include <cstdio>
+#include <libintl.h>
+#include <string.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/nebula.cpp celestia-1.5.0/src/celengine/nebula.cpp
--- celestia-1.5.0-upstream/src/celengine/nebula.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/nebula.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -21,6 +21,7 @@
 #include "glext.h"
 #include "vecgl.h"
 #include "render.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/opencluster.cpp celestia-1.5.0/src/celengine/opencluster.cpp
--- celestia-1.5.0-upstream/src/celengine/opencluster.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/opencluster.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -19,6 +19,7 @@
 #include "gl.h"
 #include "vecgl.h"
 #include "render.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/overlay.cpp celestia-1.5.0/src/celengine/overlay.cpp
--- celestia-1.5.0-upstream/src/celengine/overlay.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/overlay.cpp	2008-03-01 08:50:23.000000000 +0100
@@ -9,6 +9,7 @@
 
 #include <cstdarg>
 #include <cstdio>
+#include <cstring>
 #include <celutil/utf8.h>
 #include "gl.h"
 #include "vecgl.h"
diff -Naur celestia-1.5.0-upstream/src/celengine/parseobject.cpp celestia-1.5.0/src/celengine/parseobject.cpp
--- celestia-1.5.0-upstream/src/celengine/parseobject.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/parseobject.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -21,6 +21,7 @@
 #include "trajmanager.h"
 #include "rotationmanager.h"
 #include "universe.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/render.cpp celestia-1.5.0/src/celengine/render.cpp
--- celestia-1.5.0-upstream/src/celengine/render.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/render.cpp	2008-03-01 08:53:29.000000000 +0100
@@ -11,6 +11,7 @@
 #include <cstdio>
 #include <cstring>
 #include <cassert>
+#include <libintl.h>
 
 #ifndef _WIN32
 #ifndef TARGET_OS_MAC
diff -Naur celestia-1.5.0-upstream/src/celengine/rotationmanager.cpp celestia-1.5.0/src/celengine/rotationmanager.cpp
--- celestia-1.5.0-upstream/src/celengine/rotationmanager.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/rotationmanager.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -14,6 +14,7 @@
 #include <celutil/debug.h>
 #include "samporient.h"
 #include "rotationmanager.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/selection.cpp celestia-1.5.0/src/celengine/selection.cpp
--- celestia-1.5.0-upstream/src/celengine/selection.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/selection.cpp	2008-03-01 09:34:25.000000000 +0100
@@ -8,6 +8,7 @@
 // of the License, or (at your option) any later version.
 
 #include <cstdio>
+#include <libintl.h>
 #include "astro.h"
 #include "selection.h"
 
diff -Naur celestia-1.5.0-upstream/src/celengine/shadermanager.cpp celestia-1.5.0/src/celengine/shadermanager.cpp
--- celestia-1.5.0-upstream/src/celengine/shadermanager.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/shadermanager.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -15,6 +15,7 @@
 #include <iomanip>
 #include <cstdio>
 #include <cassert>
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/simulation.cpp celestia-1.5.0/src/celengine/simulation.cpp
--- celestia-1.5.0-upstream/src/celengine/simulation.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/simulation.cpp	2008-03-01 09:34:28.000000000 +0100
@@ -11,6 +11,7 @@
 // of the License, or (at your option) any later version.
 
 #include <algorithm>
+#include <libintl.h>
 #include "simulation.h"
 
 using namespace std;
diff -Naur celestia-1.5.0-upstream/src/celengine/solarsys.cpp celestia-1.5.0/src/celengine/solarsys.cpp
--- celestia-1.5.0-upstream/src/celengine/solarsys.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/solarsys.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -30,6 +30,7 @@
 #include "universe.h"
 #include "multitexture.h"
 #include "parseobject.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/solarsysxml.cpp celestia-1.5.0/src/celengine/solarsysxml.cpp
--- celestia-1.5.0-upstream/src/celengine/solarsysxml.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/solarsysxml.cpp	2008-03-01 11:52:31.000000000 +0100
@@ -8,6 +8,8 @@
 // of the License, or (at your option) any later version.
 
 #include <cassert>
+#include <cstring>
+#include <string.h>
 
 #include <libxml/parser.h>
 #include <libxml/parserInternals.h>
@@ -20,6 +22,7 @@
 #include "texmanager.h"
 #include "meshmanager.h"
 #include "solarsysxml.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/spiceinterface.cpp celestia-1.5.0/src/celengine/spiceinterface.cpp
--- celestia-1.5.0-upstream/src/celengine/spiceinterface.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/spiceinterface.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -11,6 +11,7 @@
 
 #include "SpiceUsr.h"
 #include "spiceinterface.h"
+#include <libintl.h>
 
 
 bool
diff -Naur celestia-1.5.0-upstream/src/celengine/spiceorbit.cpp celestia-1.5.0/src/celengine/spiceorbit.cpp
--- celestia-1.5.0-upstream/src/celengine/spiceorbit.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/spiceorbit.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -14,6 +14,7 @@
 #include "SpiceUsr.h"
 #include "astro.h"
 #include "spiceorbit.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/stardb.cpp celestia-1.5.0/src/celengine/stardb.cpp
--- celestia-1.5.0-upstream/src/celengine/stardb.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/stardb.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -10,6 +10,7 @@
 #include <cmath>
 #include <cstdlib>
 #include <cstdio>
+#include <cstring>
 #include <cassert>
 #include <algorithm>
 #include <celmath/mathlib.h>
@@ -24,6 +25,7 @@
 #include "multitexture.h"
 #include "meshmanager.h"
 #include <celutil/debug.h>
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/texmanager.cpp celestia-1.5.0/src/celengine/texmanager.cpp
--- celestia-1.5.0-upstream/src/celengine/texmanager.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/texmanager.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -13,6 +13,7 @@
 #include <fstream>
 #include "multitexture.h"
 #include "texmanager.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/texture.cpp celestia-1.5.0/src/celengine/texture.cpp
--- celestia-1.5.0-upstream/src/celengine/texture.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/texture.cpp	2008-03-01 08:54:00.000000000 +0100
@@ -28,6 +28,7 @@
 #include <cstdlib>
 #include <cstdio>
 #include <cassert>
+#include <libintl.h>
 
 #ifndef _WIN32
 #ifndef TARGET_OS_MAC
diff -Naur celestia-1.5.0-upstream/src/celengine/tokenizer.cpp celestia-1.5.0/src/celengine/tokenizer.cpp
--- celestia-1.5.0-upstream/src/celengine/tokenizer.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/tokenizer.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -12,6 +12,7 @@
 #include <iomanip>
 #include <celutil/utf8.h>
 #include "tokenizer.h"
+#include <libintl.h>
 
 
 static bool issep(char c)
diff -Naur celestia-1.5.0-upstream/src/celengine/trajmanager.cpp celestia-1.5.0/src/celengine/trajmanager.cpp
--- celestia-1.5.0-upstream/src/celengine/trajmanager.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/trajmanager.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -15,6 +15,7 @@
 #include <celutil/debug.h>
 #include "samporbit.h"
 #include "trajmanager.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/universe.cpp celestia-1.5.0/src/celengine/universe.cpp
--- celestia-1.5.0-upstream/src/celengine/universe.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/universe.cpp	2008-03-01 09:34:33.000000000 +0100
@@ -13,6 +13,7 @@
 #include <celmath/intersect.h>
 #include <celutil/utf8.h>
 #include <cassert>
+#include <libintl.h>
 #include "astro.h"
 #include "3dsmesh.h"
 #include "meshmanager.h"
diff -Naur celestia-1.5.0-upstream/src/celengine/vertexprog.cpp celestia-1.5.0/src/celengine/vertexprog.cpp
--- celestia-1.5.0-upstream/src/celengine/vertexprog.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/vertexprog.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -14,6 +14,7 @@
 #include "gl.h"
 #include "glext.h"
 #include "vertexprog.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celengine/virtualtex.cpp celestia-1.5.0/src/celengine/virtualtex.cpp
--- celestia-1.5.0-upstream/src/celengine/virtualtex.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celengine/virtualtex.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -19,6 +19,7 @@
 #include "virtualtex.h"
 #include "gl.h"
 #include "parser.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celestia/celestiacore.cpp celestia-1.5.0/src/celestia/celestiacore.cpp
--- celestia-1.5.0-upstream/src/celestia/celestiacore.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/celestiacore.cpp	2008-03-01 08:54:18.000000000 +0100
@@ -43,6 +43,7 @@
 #include <celutil/debug.h>
 #include <celutil/utf8.h>
 #include "url.h"
+#include <libintl.h>
 
 #ifdef CELX
 #include <celengine/scriptobject.h>
diff -Naur celestia-1.5.0-upstream/src/celestia/celx.cpp celestia-1.5.0/src/celestia/celx.cpp
--- celestia-1.5.0-upstream/src/celestia/celx.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/celx.cpp	2008-03-01 11:52:38.000000000 +0100
@@ -11,7 +11,9 @@
 
 #include <cassert>
 #include <cstring>
+#include <string.h>
 #include <cstdio>
+#include <libintl.h>
 #include <map>
 #include <celengine/astro.h>
 #include <celengine/celestia.h>
diff -Naur celestia-1.5.0-upstream/src/celestia/configfile.cpp celestia-1.5.0/src/celestia/configfile.cpp
--- celestia-1.5.0-upstream/src/celestia/configfile.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/configfile.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -16,6 +16,7 @@
 #include <celengine/celestia.h>
 #include <celengine/texmanager.h>
 #include "configfile.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celestia/destination.cpp celestia-1.5.0/src/celestia/destination.cpp
--- celestia-1.5.0-upstream/src/celestia/destination.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/destination.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -14,6 +14,7 @@
 #include <celengine/astro.h>
 #include <celengine/parser.h>
 #include "destination.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celestia/eclipsefinder.cpp celestia-1.5.0/src/celestia/eclipsefinder.cpp
--- celestia-1.5.0-upstream/src/celestia/eclipsefinder.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/eclipsefinder.cpp	2008-03-01 09:14:53.000000000 +0100
@@ -15,6 +15,7 @@
 #include <algorithm>
 #include <set>
 #include <cassert>
+#include <cstring>
 #include "eclipsefinder.h"
 #include "celmath/mathlib.h"
 #include "celmath/ray.h"
diff -Naur celestia-1.5.0-upstream/src/celestia/glutmain.cpp celestia-1.5.0/src/celestia/glutmain.cpp
--- celestia-1.5.0-upstream/src/celestia/glutmain.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/glutmain.cpp	2008-03-01 08:55:18.000000000 +0100
@@ -15,6 +15,7 @@
 #include <cstdlib>
 #include <cctype>
 #include <cstring>
+#include <libintl.h>
 #include <time.h>
 #include <unistd.h>
 #include <celengine/gl.h>
diff -Naur celestia-1.5.0-upstream/src/celestia/gtk/actions.cpp celestia-1.5.0/src/celestia/gtk/actions.cpp
--- celestia-1.5.0-upstream/src/celestia/gtk/actions.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/gtk/actions.cpp	2008-03-01 08:55:33.000000000 +0100
@@ -15,6 +15,7 @@
 #endif /* HAVE_CONFIG_H */
 
 #include <cstring>
+#include <libintl.h>
 #include <fstream>
 #include <gtk/gtk.h>
 
diff -Naur celestia-1.5.0-upstream/src/celestia/gtk/common.cpp celestia-1.5.0/src/celestia/gtk/common.cpp
--- celestia-1.5.0-upstream/src/celestia/gtk/common.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/gtk/common.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -20,6 +20,7 @@
 #include <celestia/celestiacore.h>
 
 #include "common.h"
+#include <libintl.h>
 
 
 /* Returns the offset of the timezone at date */
diff -Naur celestia-1.5.0-upstream/src/celestia/gtk/dialog-eclipse.cpp celestia-1.5.0/src/celestia/gtk/dialog-eclipse.cpp
--- celestia-1.5.0-upstream/src/celestia/gtk/dialog-eclipse.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/gtk/dialog-eclipse.cpp	2008-03-01 08:51:48.000000000 +0100
@@ -18,6 +18,7 @@
 
 #include "dialog-eclipse.h"
 #include "common.h"
+#include <libintl.h>
 
 
 /* Definitions: Callbacks */
diff -Naur celestia-1.5.0-upstream/src/celestia/gtk/dialog-goto.cpp celestia-1.5.0/src/celestia/gtk/dialog-goto.cpp
--- celestia-1.5.0-upstream/src/celestia/gtk/dialog-goto.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/gtk/dialog-goto.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -18,6 +18,7 @@
 
 #include "dialog-goto.h"
 #include "common.h"
+#include <libintl.h>
 
 
 /* Declarations: Callbacks */
diff -Naur celestia-1.5.0-upstream/src/celestia/gtk/dialog-options.cpp celestia-1.5.0/src/celestia/gtk/dialog-options.cpp
--- celestia-1.5.0-upstream/src/celestia/gtk/dialog-options.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/gtk/dialog-options.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -17,6 +17,7 @@
 #include "dialog-options.h"
 #include "common.h"
 #include "ui.h"
+#include <libintl.h>
 
 
 /* Definitions: Callbacks */
diff -Naur celestia-1.5.0-upstream/src/celestia/gtk/dialog-solar.cpp celestia-1.5.0/src/celestia/gtk/dialog-solar.cpp
--- celestia-1.5.0-upstream/src/celestia/gtk/dialog-solar.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/gtk/dialog-solar.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -22,6 +22,7 @@
 #include "dialog-solar.h"
 #include "actions.h"
 #include "common.h"
+#include <libintl.h>
 
 
 /* Declarations: Callbacks */
diff -Naur celestia-1.5.0-upstream/src/celestia/gtk/dialog-star.cpp celestia-1.5.0/src/celestia/gtk/dialog-star.cpp
--- celestia-1.5.0-upstream/src/celestia/gtk/dialog-star.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/gtk/dialog-star.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -25,6 +25,7 @@
 #include "dialog-star.h"
 #include "actions.h"
 #include "common.h"
+#include <libintl.h>
 
 
 /* Declarations: Callbacks */
diff -Naur celestia-1.5.0-upstream/src/celestia/gtk/dialog-time.cpp celestia-1.5.0/src/celestia/gtk/dialog-time.cpp
--- celestia-1.5.0-upstream/src/celestia/gtk/dialog-time.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/gtk/dialog-time.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -18,6 +18,7 @@
 
 #include "dialog-time.h"
 #include "common.h"
+#include <libintl.h>
 
 
 /* Declarations: Callbacks */
diff -Naur celestia-1.5.0-upstream/src/celestia/gtk/dialog-tour.cpp celestia-1.5.0/src/celestia/gtk/dialog-tour.cpp
--- celestia-1.5.0-upstream/src/celestia/gtk/dialog-tour.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/gtk/dialog-tour.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -18,6 +18,7 @@
 
 #include "dialog-tour.h"
 #include "common.h"
+#include <libintl.h>
 
 
 /* Declarations: Callbacks */
diff -Naur celestia-1.5.0-upstream/src/celestia/gtk/main.cpp celestia-1.5.0/src/celestia/gtk/main.cpp
--- celestia-1.5.0-upstream/src/celestia/gtk/main.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/gtk/main.cpp	2008-03-01 08:56:03.000000000 +0100
@@ -19,6 +19,7 @@
 #include <cstdlib>
 #include <cctype>
 #include <cstring>
+#include <libintl.h>
 #include <time.h>
 
 #ifdef WIN32
diff -Naur celestia-1.5.0-upstream/src/celestia/gtk/menu-context.cpp celestia-1.5.0/src/celestia/gtk/menu-context.cpp
--- celestia-1.5.0-upstream/src/celestia/gtk/menu-context.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/gtk/menu-context.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -20,6 +20,7 @@
 #include "menu-context.h"
 #include "actions.h"
 #include "common.h"
+#include <libintl.h>
 
 
 /* Definitions: Callbacks */
diff -Naur celestia-1.5.0-upstream/src/celestia/gtk/settings-file.cpp celestia-1.5.0/src/celestia/gtk/settings-file.cpp
--- celestia-1.5.0-upstream/src/celestia/gtk/settings-file.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/gtk/settings-file.cpp	2008-03-01 09:34:40.000000000 +0100
@@ -11,6 +11,7 @@
  */
 
 #include <gtk/gtk.h>
+#include <libintl.h>
 
 #include <celengine/body.h>
 #include <celengine/galaxy.h>
diff -Naur celestia-1.5.0-upstream/src/celestia/gtk/settings-gconf.cpp celestia-1.5.0/src/celestia/gtk/settings-gconf.cpp
--- celestia-1.5.0-upstream/src/celestia/gtk/settings-gconf.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/gtk/settings-gconf.cpp	2008-03-01 09:34:44.000000000 +0100
@@ -12,6 +12,7 @@
 
 #include <gtk/gtk.h>
 #include <gconf/gconf-client.h>
+#include <libintl.h>
 
 #include <celengine/body.h>
 #include <celengine/galaxy.h>
diff -Naur celestia-1.5.0-upstream/src/celestia/gtk/splash.cpp celestia-1.5.0/src/celestia/gtk/splash.cpp
--- celestia-1.5.0-upstream/src/celestia/gtk/splash.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/gtk/splash.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -22,6 +22,7 @@
 
 #include "splash.h"
 #include "common.h"
+#include <libintl.h>
 
 
 /* Declarations */
diff -Naur celestia-1.5.0-upstream/src/celestia/kde/celestialbrowser.cpp celestia-1.5.0/src/celestia/kde/celestialbrowser.cpp
--- celestia-1.5.0-upstream/src/celestia/kde/celestialbrowser.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/kde/celestialbrowser.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -16,6 +16,7 @@
 #include "cellistviewitem.h"
 #include "selectionpopup.h"
 #include "celutil/utf8.h"
+#include <libintl.h>
 
 /*
  *  Constructs a CelestialBrowser which is a child of 'parent', with the
diff -Naur celestia-1.5.0-upstream/src/celestia/kde/celsplashscreen.cpp celestia-1.5.0/src/celestia/kde/celsplashscreen.cpp
--- celestia-1.5.0-upstream/src/celestia/kde/celsplashscreen.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/kde/celsplashscreen.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -22,6 +22,7 @@
 #include <kglobalsettings.h>
 #include <celsplashscreen.h>
 #include <X11/Xlib.h>
+#include <libintl.h>
 
 CelSplashScreen::CelSplashScreen(const QString& filename, QWidget* _parent):
     QWidget(0, 0, WStyle_Customize | WX11BypassWM), 
diff -Naur celestia-1.5.0-upstream/src/celestia/kde/eclipsefinderdlg.cpp celestia-1.5.0/src/celestia/kde/eclipsefinderdlg.cpp
--- celestia-1.5.0-upstream/src/celestia/kde/eclipsefinderdlg.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/kde/eclipsefinderdlg.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -11,6 +11,7 @@
 #include "celestiacore.h"
 #include "celengine/astro.h"
 #include "eclipsefinder.h"
+#include <libintl.h>
 
 /* 
  *  Constructs a EclipseFinder which is a child of 'parent', with the 
diff -Naur celestia-1.5.0-upstream/src/celestia/kde/kcelbookmarkmanager.cpp celestia-1.5.0/src/celestia/kde/kcelbookmarkmanager.cpp
--- celestia-1.5.0-upstream/src/celestia/kde/kcelbookmarkmanager.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/kde/kcelbookmarkmanager.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -19,6 +19,7 @@
 #include <qdir.h>
 #include <kstandarddirs.h>
 #include "kcelbookmarkmanager.h"
+#include <libintl.h>
 
 KBookmarkManager* KCelBookmarkManager::self() {
     if ( !s_bookmarkManager )
diff -Naur celestia-1.5.0-upstream/src/celestia/kde/kcelbookmarkmenu.cpp celestia-1.5.0/src/celestia/kde/kcelbookmarkmenu.cpp
--- celestia-1.5.0-upstream/src/celestia/kde/kcelbookmarkmenu.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/kde/kcelbookmarkmenu.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -45,6 +45,7 @@
 #include <kpopupmenu.h>
 #include <kstdaccel.h>
 #include <kstdaction.h>
+#include <libintl.h>
 
 template class QPtrList<KCelBookmarkMenu>;
 
diff -Naur celestia-1.5.0-upstream/src/celestia/kde/kdeapp.cpp celestia-1.5.0/src/celestia/kde/kdeapp.cpp
--- celestia-1.5.0-upstream/src/celestia/kde/kdeapp.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/kde/kdeapp.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -87,6 +87,7 @@
 #endif
 
 #include "celengine/glext.h"
+#include <libintl.h>
 
 #define MENUMAXSIZE 100
 
diff -Naur celestia-1.5.0-upstream/src/celestia/kde/kdeglwidget.cpp celestia-1.5.0/src/celestia/kde/kdeglwidget.cpp
--- celestia-1.5.0-upstream/src/celestia/kde/kdeglwidget.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/kde/kdeglwidget.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -48,6 +48,7 @@
 
 #include <math.h>
 #include <vector>
+#include <libintl.h>
 
 KdeGlWidget::KdeGlWidget(  QWidget* parent, const char* name, CelestiaCore* core)
     : QGLWidget( parent, name )
diff -Naur celestia-1.5.0-upstream/src/celestia/kde/kdemain.cpp celestia-1.5.0/src/celestia/kde/kdemain.cpp
--- celestia-1.5.0-upstream/src/celestia/kde/kdemain.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/kde/kdemain.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -20,6 +20,7 @@
 #include <klocale.h>
 
 #include "kdeuniquecelestia.h"
+#include <libintl.h>
 
 static const char *description =
     I18N_NOOP("Celestia");
diff -Naur celestia-1.5.0-upstream/src/celestia/kde/kdepreferencesdialog.cpp celestia-1.5.0/src/celestia/kde/kdepreferencesdialog.cpp
--- celestia-1.5.0-upstream/src/celestia/kde/kdepreferencesdialog.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/kde/kdepreferencesdialog.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -40,6 +40,7 @@
 #include "celengine/render.h"
 #include "celengine/glcontext.h"
 #include "celengine/astro.h"
+#include <libintl.h>
 
 static uint32 FilterOtherLocations = ~(Location::City |
                     Location::Observatory |
diff -Naur celestia-1.5.0-upstream/src/celestia/kde/kdeuniquecelestia.cpp celestia-1.5.0/src/celestia/kde/kdeuniquecelestia.cpp
--- celestia-1.5.0-upstream/src/celestia/kde/kdeuniquecelestia.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/kde/kdeuniquecelestia.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -21,6 +21,7 @@
 #include <string>
 #include <vector>
 #include <klocale.h>
+#include <libintl.h>
 
 KdeUniqueCelestia::KdeUniqueCelestia() {
 
diff -Naur celestia-1.5.0-upstream/src/celestia/kde/selectionpopup.cpp celestia-1.5.0/src/celestia/kde/selectionpopup.cpp
--- celestia-1.5.0-upstream/src/celestia/kde/selectionpopup.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/kde/selectionpopup.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -29,6 +29,7 @@
 #include <krun.h>
 
 #include "celutil/utf8.h"
+#include <libintl.h>
 
 SelectionPopup::SelectionPopup(QWidget* parent, CelestiaCore* _appCore, Selection _sel):
 KPopupMenu(parent),
diff -Naur celestia-1.5.0-upstream/src/celestia/kde/videocapturedlg.cpp celestia-1.5.0/src/celestia/kde/videocapturedlg.cpp
--- celestia-1.5.0-upstream/src/celestia/kde/videocapturedlg.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/kde/videocapturedlg.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -36,6 +36,7 @@
 #include <kstatusbar.h>
 #include <kapp.h>
 #include "kdeapp.h"
+#include <libintl.h>
 
 VideoCaptureDlg::VideoCaptureDlg(QWidget* p, const QString &dir):
     accepted(false)
diff -Naur celestia-1.5.0-upstream/src/celestia/oggtheoracapture.cpp celestia-1.5.0/src/celestia/oggtheoracapture.cpp
--- celestia-1.5.0-upstream/src/celestia/oggtheoracapture.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/oggtheoracapture.cpp	2008-03-01 09:15:02.000000000 +0100
@@ -66,11 +66,14 @@
 #include <celutil/util.h>
 #include "../celengine/gl.h"
 #include <string>
+#include <cstring>
+#include <ctime>
 #include "theora/theora.h"
 
 using namespace std;
 
 #include "oggtheoracapture.h"
+#include <libintl.h>
 
 //  {"video-rate-target",required_argument,NULL,'V'},
 //  {"video-quality",required_argument,NULL,'v'},
diff -Naur celestia-1.5.0-upstream/src/celestia/scriptmenu.cpp celestia-1.5.0/src/celestia/scriptmenu.cpp
--- celestia-1.5.0-upstream/src/celestia/scriptmenu.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/scriptmenu.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -13,6 +13,7 @@
 #include "celutil/directory.h"
 #include "celutil/filetype.h"
 #include <fstream>
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celestia/url.cpp celestia-1.5.0/src/celestia/url.cpp
--- celestia-1.5.0-upstream/src/celestia/url.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/url.cpp	2008-03-01 11:52:48.000000000 +0100
@@ -16,10 +16,12 @@
  ***************************************************************************/
 
 #include <string>
+#include <string.h>
 #include <stdio.h>
 #include "celestiacore.h"
 #include "celengine/astro.h"
 #include "url.h"
+#include <libintl.h>
 
 static const unsigned int CurrentCelestiaURLVersion = 2;
 
diff -Naur celestia-1.5.0-upstream/src/celestia/wglext.cpp celestia-1.5.0/src/celestia/wglext.cpp
--- celestia-1.5.0-upstream/src/celestia/wglext.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/wglext.cpp	2008-03-01 11:52:52.000000000 +0100
@@ -14,7 +14,10 @@
 #include "wglext.h"
 #include <windows.h>
 #include <cstdio>
+#include <cstring>
+#include <string.h>
 #include <vector>
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celestia/winbookmarks.cpp celestia-1.5.0/src/celestia/winbookmarks.cpp
--- celestia-1.5.0-upstream/src/celestia/winbookmarks.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/winbookmarks.cpp	2008-03-01 11:52:58.000000000 +0100
@@ -13,6 +13,7 @@
 #include "res/resource.h"
 #include <celutil/winutil.h>
 #include <iostream>
+#include <string.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celestia/windatepicker.cpp celestia-1.5.0/src/celestia/windatepicker.cpp
--- celestia-1.5.0-upstream/src/celestia/windatepicker.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/windatepicker.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -11,10 +11,12 @@
 
 #include <windows.h>
 #include <commctrl.h>
+#include <cstring>
 #include "celutil/basictypes.h"
 #include "celengine/astro.h"
 #include "celutil/util.h"
 #include "celutil/winutil.h"
+#include <libintl.h>
 
 
 // DatePicker is a Win32 control for setting the date. It replaces the
diff -Naur celestia-1.5.0-upstream/src/celestia/wineclipses.cpp celestia-1.5.0/src/celestia/wineclipses.cpp
--- celestia-1.5.0-upstream/src/celestia/wineclipses.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/wineclipses.cpp	2008-03-01 11:53:02.000000000 +0100
@@ -10,6 +10,7 @@
 
 #include <string>
 #include <sstream>
+#include <string.h>
 #include <algorithm>
 #include <set>
 #include <cassert>
@@ -23,6 +24,7 @@
 #include "celmath/distance.h"
 #include "celutil/util.h"
 #include "celutil/winutil.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celestia/winhyperlinks.cpp celestia-1.5.0/src/celestia/winhyperlinks.cpp
--- celestia-1.5.0-upstream/src/celestia/winhyperlinks.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/winhyperlinks.cpp	2008-03-01 08:50:24.000000000 +0100
@@ -9,6 +9,7 @@
 // as published by the Free Software Foundation; either version 2
 // of the License, or (at your option) any later version.
 
+#include <cstring>
 #include "winhyperlinks.h"
 #include "res/resource.h"
 
diff -Naur celestia-1.5.0-upstream/src/celestia/winmain.cpp celestia-1.5.0/src/celestia/winmain.cpp
--- celestia-1.5.0-upstream/src/celestia/winmain.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/winmain.cpp	2008-03-01 11:53:07.000000000 +0100
@@ -23,6 +23,7 @@
 #include <windows.h>
 #include <commctrl.h>
 #include <mmsystem.h>
+#include <string.h>
 
 #include <celmath/vecmath.h>
 #include <celmath/quaternion.h>
@@ -59,6 +60,7 @@
 #include "wglext.h"
 
 #include <locale.h>
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celestia/winsplash.cpp celestia-1.5.0/src/celestia/winsplash.cpp
--- celestia-1.5.0-upstream/src/celestia/winsplash.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/winsplash.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -15,8 +15,9 @@
 #include <commctrl.h>
 #include "res/resource.h"
 #include "celutil/util.h"
 #include <iostream>
+#include <libintl.h>
 #include <locale.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celestia/winstarbrowser.cpp celestia-1.5.0/src/celestia/winstarbrowser.cpp
--- celestia-1.5.0-upstream/src/celestia/winstarbrowser.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/winstarbrowser.cpp	2008-03-01 11:53:13.000000000 +0100
@@ -15,10 +15,12 @@
 #include <windows.h>
 #include <commctrl.h>
 #include <cstring>
+#include <string.h>
 #include "winstarbrowser.h"
 #include "celutil/winutil.h"
 
 #include "res/resource.h"
+#include <libintl.h>
 
 extern void SetMouseCursor(LPCTSTR lpCursor);
 
diff -Naur celestia-1.5.0-upstream/src/celestia/wintime.cpp celestia-1.5.0/src/celestia/wintime.cpp
--- celestia-1.5.0-upstream/src/celestia/wintime.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celestia/wintime.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -18,6 +18,7 @@
 #include <celengine/astro.h>
 #include "celutil/util.h"
 #include "celutil/winutil.h"
+#include <libintl.h>
 
 
 
diff -Naur celestia-1.5.0-upstream/src/celmath/quaternion.h celestia-1.5.0/src/celmath/quaternion.h
--- celestia-1.5.0-upstream/src/celmath/quaternion.h	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celmath/quaternion.h	2008-03-01 12:30:07.000000000 +0100
@@ -16,7 +16,6 @@
 #include <celmath/mathlib.h>
 #include <celmath/vecmath.h>
 
-
 template<class T> class Quaternion
 {
 public:
diff -Naur celestia-1.5.0-upstream/src/celutil/bigfix.cpp celestia-1.5.0/src/celutil/bigfix.cpp
--- celestia-1.5.0-upstream/src/celutil/bigfix.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celutil/bigfix.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -10,6 +10,7 @@
 #include <math.h>
 #include <stdio.h>
 #include "bigfix.h"
+#include <libintl.h>
 
 
 /*** Constructors ***/
diff -Naur celestia-1.5.0-upstream/src/celutil/color.cpp celestia-1.5.0/src/celutil/color.cpp
--- celestia-1.5.0-upstream/src/celutil/color.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celutil/color.cpp	2008-03-01 11:53:16.000000000 +0100
@@ -9,6 +9,7 @@
 
 #include <cstdio>
 #include <cstring>
+#include <string.h>
 #include <cctype>
 #include "color.h"
 
diff -Naur celestia-1.5.0-upstream/src/celutil/directory.cpp celestia-1.5.0/src/celutil/directory.cpp
--- celestia-1.5.0-upstream/src/celutil/directory.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celutil/directory.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -9,6 +9,7 @@
 
 #include <iostream>
 #include "directory.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celutil/filetype.cpp celestia-1.5.0/src/celutil/filetype.cpp
--- celestia-1.5.0-upstream/src/celutil/filetype.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celutil/filetype.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -11,6 +11,7 @@
 #include <cstdlib>
 #include "util.h"
 #include "filetype.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celutil/formatnum.cpp celestia-1.5.0/src/celutil/formatnum.cpp
--- celestia-1.5.0-upstream/src/celutil/formatnum.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celutil/formatnum.cpp	2008-03-01 08:50:23.000000000 +0100
@@ -10,6 +10,8 @@
 #include <cmath>
 #include <cstdio>
 #include <string>
+#include <cstring>
+#include <limits.h>
 #include "formatnum.h"
 
 // HACK: MS Visual C++ has _snprintf declared in stdio.h but not snprintf
diff -Naur celestia-1.5.0-upstream/src/celutil/utf8.cpp celestia-1.5.0/src/celutil/utf8.cpp
--- celestia-1.5.0-upstream/src/celutil/utf8.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celutil/utf8.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -9,7 +9,9 @@
 
 #include "utf8.h"
 #include <cctype>
+#include <cstring>
 #include "util.h"
+#include <libintl.h>
 
 unsigned int WGL4_Normalization_00[256] = {
     0x0000, 0x0001, 0x0002, 0x0003, 0x0004, 0x0005, 0x0006, 0x0007,
diff -Naur celestia-1.5.0-upstream/src/celutil/util.cpp celestia-1.5.0/src/celutil/util.cpp
--- celestia-1.5.0-upstream/src/celutil/util.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celutil/util.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -12,6 +12,7 @@
 #include "util.h"
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/celutil/util.h celestia-1.5.0/src/celutil/util.h
--- celestia-1.5.0-upstream/src/celutil/util.h	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celutil/util.h	2008-03-01 11:49:36.000000000 +0100
@@ -22,11 +22,11 @@
     switch(0){case 0: case pred:;}
 
 #ifndef min
-#define min(a, b) ((a) < (b) ? (a) : (b))
+template <typename T1, typename T2> bool min(T1 a, T2 b){ return a<(T1)b; }
 #endif
 
 #ifndef max
-#define max(a, b) ((a) < (b) ? (b) : (a))
+template <typename T1, typename T2> bool max(T1 a, T2 b){ return a<(T1)b; }
 #endif
 				      
 #ifdef _WIN32
diff -Naur celestia-1.5.0-upstream/src/celutil/windirectory.cpp celestia-1.5.0/src/celutil/windirectory.cpp
--- celestia-1.5.0-upstream/src/celutil/windirectory.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/celutil/windirectory.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -10,6 +10,7 @@
 #include <iostream>
 #include <windows.h>
 #include "directory.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/packdb.cpp celestia-1.5.0/src/packdb.cpp
--- celestia-1.5.0-upstream/src/packdb.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/packdb.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -10,6 +10,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
+#include <libintl.h>
 
 #define SPECTRAL_O 0
 #define SPECTRAL_B 1
diff -Naur celestia-1.5.0-upstream/src/tools/atmosphere/scattersim.cpp celestia-1.5.0/src/tools/atmosphere/scattersim.cpp
--- celestia-1.5.0-upstream/src/tools/atmosphere/scattersim.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/tools/atmosphere/scattersim.cpp	2008-03-01 11:53:23.000000000 +0100
@@ -11,7 +11,9 @@
 #include <iostream>
 #include <fstream>
 #include <string>
+#include <string.h>
 #include <cstdlib>
+#include <libintl.h>
 #include <cmath>
 #include <algorithm>
 #include <map>
diff -Naur celestia-1.5.0-upstream/src/tools/buildstardb.cpp celestia-1.5.0/src/tools/buildstardb.cpp
--- celestia-1.5.0-upstream/src/tools/buildstardb.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/tools/buildstardb.cpp	2008-03-01 08:51:49.000000000 +0100
@@ -16,6 +16,7 @@
 #include <cstdio>
 #include <assert.h>
 #include "stardb.h"
+#include <libintl.h>
 
 using namespace std;
 
diff -Naur celestia-1.5.0-upstream/src/tools/cmod/cmodfix.cpp celestia-1.5.0/src/tools/cmod/cmodfix.cpp
--- celestia-1.5.0-upstream/src/tools/cmod/cmodfix.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/tools/cmod/cmodfix.cpp	2008-03-01 11:53:27.000000000 +0100
@@ -18,6 +18,7 @@
 #include <cassert>
 #include <cmath>
 #include <cstdio>
+#include <string.h>
 #include <algorithm>
 #include <vector>
 #ifdef TRISTRIP
diff -Naur celestia-1.5.0-upstream/src/tools/cmod/xtocmod.cpp celestia-1.5.0/src/tools/cmod/xtocmod.cpp
--- celestia-1.5.0-upstream/src/tools/cmod/xtocmod.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/tools/cmod/xtocmod.cpp	2008-03-01 11:53:31.000000000 +0100
@@ -3,6 +3,7 @@
 #include <iomanip>
 #include <algorithm>
 #include <cstdio>
+#include <string.h>
 #include <cassert>
 #include <cstring>
 #include <d3dx9.h>
diff -Naur celestia-1.5.0-upstream/src/tools/stardb/makestardb.cpp celestia-1.5.0/src/tools/stardb/makestardb.cpp
--- celestia-1.5.0-upstream/src/tools/stardb/makestardb.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/tools/stardb/makestardb.cpp	2008-03-01 11:53:34.000000000 +0100
@@ -14,6 +14,7 @@
 #include <iomanip>
 #include <cctype>
 #include <cassert>
+#include <string.h>
 #include <celutil/basictypes.h>
 #include <celutil/bytes.h>
 #include <celengine/astro.h>
diff -Naur celestia-1.5.0-upstream/src/tools/stardb/startextdump.cpp celestia-1.5.0/src/tools/stardb/startextdump.cpp
--- celestia-1.5.0-upstream/src/tools/stardb/startextdump.cpp	2008-03-01 12:51:20.000000000 +0100
+++ celestia-1.5.0/src/tools/stardb/startextdump.cpp	2008-03-01 11:53:40.000000000 +0100
@@ -13,6 +13,7 @@
 #include <iostream>
 #include <fstream>
 #include <iomanip>
+#include <string.h>
 #include <celutil/basictypes.h>
 #include <celutil/bytes.h>
 #include <celengine/astro.h>
